src/libPMacc/include/random/distributions/uniform/Uniform_float.hpp

```diff
@@ -48,10 +54,125 @@ namespace detail
+            const result_type 2pow32Inv = 2.3283064e-10f;
+            const uint32_t random = RNGMethod().get32Bits(state);
+            return static_cast<result_type>( random ) * 2pow32Inv +
+                ( 2pow32Inv / 2.0f );
         }
     };
 
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Swap the value one to zero (creates a small error in uniform distribution)
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::SwapOneToZero,
+        T_RNGMethod,
+        void
+    >
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            const result_type randomValue =
+                PMacc::random::distributions::Uniform<
+                    uniform::ExcludeZero<result_type>
+            >()(state);
+            return randomValue == 1.0f ? 0.0f : randomValue;
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Reduce the random range to 2^24.
+     * Creates intervals with the width of epsilon/2.
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::Use24Bit,
+        T_RNGMethod,
+        void
+    >
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            const result_type 2pow24Inv = 5.9604645e-08f;
+            const randomValue24Bit = RNGMethod().get32Bits(state) >> 8;
+            return static_cast<result_type>( randomValue24Bit ) * PMACC_RNG_2POW24_INV;
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Loops until a random value inside the defined range is created.
+     * The runtime of this method is not deterministic.
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::Repeat,
+        T_RNGMethod,
+        void
+    >;
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            do
+            {
+                const result_type randomValue =
+                    PMacc::random::distributions::Uniform<
+                        uniform::ExcludeZero<result_type>
+                    >()(state);
+                if( randomValue != 1.0f )
+                    return randomValue;
+            }
+            while(true);
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * The default case use the method
```

Comment:
Just write here: `Equivalent to uniform::ExcludeOne<float>::Use24Bit` or better (IMO): Unspecified method is used for corner cases because this is the default for people only interested in getting a uniform float in `[0,1)` not caring how. So leave us the freedom to choose an implementation here.